transaction/core/src/ring_signature/key_image.rs

```diff
@@ -10,87 +10,97 @@ use serde::{Deserialize, Serialize};
 
 use super::Error;
 use crate::ring_signature::Scalar;
+use blake2::Blake2b;
 use core::convert::TryInto;
 use curve25519_dalek::constants::RISTRETTO_BASEPOINT_POINT;
+use keys::{RistrettoPrivate, RistrettoPublic};
 
 #[derive(Copy, Clone, Default, Eq, Serialize, Deserialize, Digestible)]
-/// The "image" of a private key `x`: I = x * H(x * G) = x * H(P).
-pub struct KeyImage(pub(crate) CompressedRistretto);
+/// The "image" of a private key `x`: I = x * Hp(x * G) = x * Hp(P).
+pub struct KeyImage {
+    pub point: CompressedRistretto,
```

Comment:
I changed this from a tuple to a struct because I find `key_image.point` easier to think about than `key_image.0`.